clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -1210,7 +1210,8 @@ mlir::Value ScalarExprEmitter::buildSub(const BinOpInfo &Ops) {
   // See more in `EmitSub` in CGExprScalar.cpp.
   assert(!UnimplementedFeature::llvmLoweringPtrDiffConsidersPointee());
   return Builder.create<mlir::cir::PtrDiffOp>(CGF.getLoc(Ops.Loc),
-                                              CGF.PtrDiffTy, Ops.LHS, Ops.RHS);
+                                              CGF.getCIRType(Ops.Ty), Ops.LHS,
```

Comment:
Another option is to change `CIRGenTypeCache::PtrDiffTy` to be distinct from `UIntPtrTy` and signed. That approach would keep code more similar to original codegen.

Reply:
I prefer the "other option": change to make it distinct from `UIntPtrTy`, since it could allow future ported code that depends on `PtrDiffTy` to just work.

Reply:
Got it, fixed.